### PR TITLE
Add support for dilated Conv1D and Conv2D

### PR DESCRIPTION
**Pull request by jon-tow**
#274
 
The documentation for some layers seem to be lacking in detail. I'll try to clarify some things when I get the chance.

### CODE REVIEW

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 1]
```diff
@@ -60,8 +65,11 @@ public struct Conv1D<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output `[batchCount, newWidth, outputChannels]`.
     @differentiable
     public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
-        let conv = conv2D(input.expandingShape(at: 1), filter: filter.expandingShape(at: 0),
-                          strides: (1, 1, stride, 1), padding: padding)
+        let conv = conv2D(input.expandingShape(at: 1),
```
Comment (eaplatanios) [inline Sources/TensorFlow/Layers/Convolutional.swift]:
```suggestion
        let conv = conv2D(
            input.expandingShape(at: 1),
```

Can you align the arguments in a new line, like in the suggested edit?

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 2]
```diff
@@ -178,8 +197,11 @@ public struct Conv2D<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output.
     @differentiable
     public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
-        return activation(conv2D(input, filter: filter, strides: (1, strides.0, strides.1, 1),
-                                 padding: padding) + bias)
+        return activation(conv2D(input,
```
Comment (eaplatanios) [inline Sources/TensorFlow/Layers/Convolutional.swift]: Same comment as above.

**Comment by jon-tow** [comment]
@eaplatanios Thanks for pointing that out. Wish we had a designated formatter :smile:.

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 3]
```diff
@@ -178,8 +198,12 @@ public struct Conv2D<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output.
     @differentiable
     public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
-        return activation(conv2D(input, filter: filter, strides: (1, strides.0, strides.1, 1),
-                                 padding: padding) + bias)
+        return activation(conv2D(
+                            input,
```
Comment (eaplatanios) [inline Sources/TensorFlow/Layers/Convolutional.swift]: The indentation for input should just be 4-spaces further than the `return` statement.

Comment (jon-tow) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Sorry about that. I was unsure what to do and should've asked beforehand.

**Comment by Shashi456** [comment]
@eaplatanios @rxwei Although I see the extensions have dilation as optional, I'm wondering why dilation wasn't made optional in the main public init, because the dilated convolutions isn't used by everyone. Even within the raw bindings, they are an optional parameter with some default value declared. Wdyt?

**Comment by eaplatanios** [comment]
@Shashi456 good point. I added some defaults along with some minor stylistic changes in #277.

**Comment by rxwei** [comment]
+1 to default arguments.

**Pull request merged**